kernel/src/process.rs

```diff
@@ -163,13 +165,22 @@ pub trait ProcessType {
 
     // mpu
 
-    fn setup_mpu(&self, mpu: &mpu::MPU);
+    /// Configure the MPU to use the process's allocated regions.
+    fn setup_mpu(&self);
 
-    fn add_mpu_region(&self, base: *const u8, size: u32) -> bool;
+    /// Allocate a new MPU region for the process that is at least `min_region_size`
+    /// bytes and lies within the specified stretch of unallocated memory.
+    fn add_mpu_region(
+        &self,
+        unallocated_memory_start: *const u8,
+        unallocated_memory_size: usize,
+        min_region_size: usize,
+    ) -> Option<(*const u8, usize)>;
```

Comment:
I think it makes sense to add something like

```rust
struct MpuRegion {
  start_address: *const u8,
  length_bytes: usize,
}
```

to be used in the return value to provide a helping hint on what the tuple is. This would also help with the mpu regions array in the process struct.